chalicelib/checks/deployment_checks.py

```diff
@@ -216,21 +216,46 @@ def deploy_application_to_beanstalk(connection, **kwargs):
     return _deploy_application_to_beanstalk(connection, **kwargs)
 
 
-@check_function()
-def deploy_ff_staging(connection, **kwargs):
-    """ Deploys Fourfront master to whoever staging is.
-        Runs as part of the 'deployment_checks' schedule on data ONLY.
+def deploy_env(connection, env_to_deploy, application_name, check, **kwargs):
+    """ For a given beanstalk environment, with a configured check json and given application name,
+        deploys the environment, or returns an error.
     """
-    this_check = CheckResult(connection, 'deploy_ff_staging')
-    env_to_deploy = compute_ff_stg_env()
+    this_check = CheckResult(connection, check)
     helper_check = _deploy_application_to_beanstalk(connection,
                                                     env=env_to_deploy,
-                                                    branch='master')
+                                                    branch='master',
+                                                    **kwargs)
     if helper_check.status == 'PASS':
         this_check.status = 'PASS'
-        this_check.summary = 'Successfully deployed Fourfront master to %s' % env_to_deploy
+        this_check.summary = 'Successfully deployed {0} master to {1}'.format(application_name, env_to_deploy)
```

Comment:
Minor: I have a very small preference, once you have multiple args, for named arguments to format, which is easier to keep track of visually than numbers, as in (untested):
```
this_check.summary = ('Successfully deployed {what} master to {where}'
                      .format(what=application_name, where=env_to_deploy))
```